dll/win32/shell32/folders/CDrivesFolder.cpp

```diff
@@ -511,18 +511,48 @@ HRESULT CDrivesExtractIcon_CreateInstance(IShellFolder * psf, LPCITEMIDLIST pidl
         DriveType = DRIVE_FIXED;
 
     WCHAR wTemp[MAX_PATH];
-    int icon_idx;
+    int icon_idx, reg_idx;
     UINT flags = 0;
-    if ((DriveType == DRIVE_FIXED || DriveType == DRIVE_UNKNOWN) &&
-        (HCR_GetIconW(L"Drive", wTemp, NULL, MAX_PATH, &icon_idx)))
+    switch (DriveType)
     {
-        initIcon->SetNormalIcon(wTemp, icon_idx);
+        case DRIVE_FIXED:
+        case DRIVE_UNKNOWN:
+            reg_idx = IDI_SHELL_DRIVE - 1;
```

Comment:
What is the meaning of minus one?

Reply:
The constant index is 1-based, while the needed value is zero based.

Reply:
Thanks.